File: frontend/crd/src/domain/Skill.ts
export interface SkillJSON{
    id: string;
    name: string;
    language: boolean;
    studentDetailsID: string;
}

export class Skill{

    constructor(json: SkillJSON){
        this.id = json.id;
        this.name = json.name;
        this.isLanguage = json.language;
        this.studentDetailsID = json.studentDetailsID;
    }

    id: string;
    name: string;
    isLanguage: boolean;
    studentDetailsID: string;

    static makeEmpty(){
        return new Skill({
            id: '',
            name: '',
            language: false,
            studentDetailsID: ''
        })
    }
}

File: backend/src/main/java/com/senior/project/backend/domain/Skill.java
package com.senior.project.backend.domain;

import lombok.*;

import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity
@Data
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Generated
public class Skill {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;
    private boolean isLanguage;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="student_details_id")
    @JsonIgnore
    @ToString.Exclude
    private StudentDetails studentDetails;
}
